File: ios/Classes/PlatformProxyPlugin.m
#import "PlatformProxyPlugin.h"

@implementation PlatformProxyPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterMethodChannel* channel = [FlutterMethodChannel
                                     methodChannelWithName:@"platform_proxy"
                                     binaryMessenger:[registrar messenger]];
    PlatformProxyPlugin* instance = [[PlatformProxyPlugin alloc] init];
    [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
    if ([@"getPlatformProxy" isEqualToString:call.method]) {
        NSString *targetUrl = [call arguments][@"url"];
        if (targetUrl) {
            YSFPPProxiesResolver *resolver = [[YSFPPProxiesResolver alloc] init];
            [resolver resolve:targetUrl];
            result([resolver proxiesAsJson]);
        } else {
            result(@"[]");
        }
    } else {
        result(FlutterMethodNotImplemented);
    }
}

@end

File: ios/Classes/PlatformProxyPlugin.swift
// PlatformProxyPlugin.swift
// Swift version of PlatformProxyPlugin for iOS
import Flutter
import UIKit

public class PlatformProxyPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "platform_proxy", binaryMessenger: registrar.messenger())
        let instance = PlatformProxyPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }
    
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getPlatformProxy":
            if let args = call.arguments as? [String: String], let url = args["url"] {
                let resolver = YSFPPProxiesResolver()
                resolver.resolve(url)
                result(resolver.proxiesAsJson())
            } else {
                result("[]")
            }
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
